skglm/tests/test_penalties.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import numpy as np
+
+from numpy.linalg import norm
+from numpy.testing import assert_array_less
+
+from skglm.datafits import Quadratic, QuadraticMultiTask
+from skglm.penalties import (
+    L1, L1_plus_L2, WeightedL1, MCPenalty, SCAD, IndicatorBox, L0_5, L2_3,
+    L2_1, L2_05, BlockMCPenalty, BlockSCAD)
+from skglm import GeneralizedLinearEstimator
+from skglm.utils import make_correlated_data
+
+
+n_samples = 20
+n_features = 10
+n_tasks = 10
+X, Y, _ = make_correlated_data(
+    n_samples=n_samples, n_features=n_features, n_tasks=n_tasks, density=0.1,
+    random_state=0)
+y = Y[:, 0]
+
+n_samples, n_features = X.shape
+alpha_max = norm(X.T @ y, ord=np.inf) / n_samples
+alpha = alpha_max / 1000
+
+penalties = [
+    L1(alpha=alpha),
+    L1_plus_L2(alpha=alpha, l1_ratio=0.5),
+    WeightedL1(alpha=1, weights=np.arange(n_features)),
+    MCPenalty(alpha=alpha, gamma=4),
+    SCAD(alpha=alpha, gamma=4),
+    IndicatorBox(alpha=alpha),
+    L0_5(alpha),
+    L2_3(alpha)]
+
+block_penalties = [
+    L2_1(alpha=alpha), L2_05(alpha=alpha),
+    BlockMCPenalty(alpha=alpha, gamma=4),
+    BlockSCAD(alpha=alpha, gamma=4)
+    ]
+
+
+@pytest.mark.parametrize('penalty', penalties)
+def test_subdiff_diff(penalty):
+    estimator_ours = GeneralizedLinearEstimator(
```

Comment:
call it clf for simplicity ? as you wish

Reply:
I went for est (for me clf stands for classifier)

Reply:
perfect